ui/src/app/nav.tsx

```diff
@@ -24,12 +33,20 @@ export function Aside({ isOpen, setIsOpen }: { isOpen: boolean, setIsOpen: Dispa
                         setCSRPEMString(e.target.result.toString());
                     }
                 }
-        };
-        reader.readAsText(file);
+            };
+            reader.readAsText(file);
+        }
+    };
+    const CSRValidationElement = function () {
+        try {
+            extractCSR(CSRPEMString)
+        } catch (e) {
+            return <div><i className="p-icon--error"></i>Invalid CSR</div>
```

Comment:
I dislike that we see the red "Invalid CSR" before even inputting the CSR. It makes me feel like I did something wrong when I didn't do anyhting yet. We could display this message only when something is inputted in the box.

Reply:
Done